src/domain/mongoose_domain_sql.erl

```diff
@@ -138,34 +143,51 @@ select_from(FromId, Limit) ->
 select_updates_from(FromId, Limit) ->
     Pool = get_db_pool(),
     Args = rdbms_queries:add_limit_arg(Limit, [FromId]),
-    {selected, Rows} = execute_successfully(Pool, domain_select_events_from, Args),
-    Rows.
-
-get_min_event_id() ->
-    Pool = get_db_pool(),
-    selected_to_int(execute_successfully(Pool, domain_events_min, [])).
+    case execute_successfully(Pool, domain_select_events_from, Args) of
+        {selected, []} ->
+            %% get MaxID, this inserts the dummy record
+            %% in the events table if required.
+            MaxId = get_max_event_id_or_set_dummy(),
+            if
+                MaxId > FromId ->
+                    select_updates_from(FromId, Limit);
+                true ->
+                    %% This must never happen though, unless the events table
+                    %% is modified externally. this is critical error!
+                    %% TODO: figure out what should we do in this case
+                    ?LOG_ERROR(#{what => select_updates_from_failed, limit => Limit,
+                                 from_id => FromId, max_id => MaxId}),
+                    []
+            end;
+        {selected, Rows} -> Rows
+    end.
 
-get_max_event_id() ->
+get_max_event_id_or_set_dummy() ->
     Pool = get_db_pool(),
-    selected_to_int(execute_successfully(Pool, domain_events_max, [])).
+    case execute_successfully(Pool, domain_events_max, []) of
+        {selected, [{null}]} ->
+            %% ensure that we have at least one record
+            %% in the table, even if it's dummy one.
+            insert_domain_event(Pool, <<"dummy.test.domain">>),
+            get_max_event_id_or_set_dummy();
+        NonNullSelection -> selected_to_int(NonNullSelection)
+    end.
 
 delete_events_older_than(Id) ->
     transaction(fun(Pool) ->
-            MaxId = get_max_event_id(),
-            if MaxId =:= 0 ->
-                    skipped;
-               MaxId < Id ->
+            MaxId = get_max_event_id_or_set_dummy(),
+            if
+                MaxId < Id ->
                     %% Removal would erase all the events, which we don't want.
                     %% We want to keep at least one event.
-                    %% This should never happen though, unless the events table
-                    %% is modified externally (for example, after DB restored
-                    %% from the dump to the previous state).
-                    %% Skipping is not critical.
+                    %% This must never happen though, unless the events table
+                    %% is modified externally. this is critical error!
+                    %% TODO: figure out what should we do in this case
```

Comment:
Actually I would just crash in this case. I don't think we need protection against someone calling `TRUNCATE TABLE` or something like that while the system is running. We can keep this code, no problem from me, we just can't try to fix all such cases.

Reply:
actually we must restart the service and make a complete reload of the table in this case, but I wrote this comment before introduction of service restart procedure.